### ISSUE TITLE
[ANDROID] getProducts never returns.

### ISSUE THREAD

**Issue by johanvdb** [j3k0/cordova-plugin-purchase#820]
I am unable to get the IAP to work with my Ionic 4 application. I have a paid developer google account, with two products created on the app store, and I am able to upload and install my app and distribute it via alpha. Everything else works, but the in app payments just hangs waiting for the promise from getProducts to return, but it never does.

I have been unable to retrieve the logs from the device (no debug in app store, and release needed to test IAP). When I run the app in an emulator, I get the same response. The debug log does not show any error or further info from the IAP. I assume that it should not work on an emulator, but am I not at least supposed to receive an error message when that is the case?

https://ionicframework.com/docs/native/in-app-purchase

### system info

Ionic:

   ionic (Ionic CLI)             : 4.12.0 
   Ionic Framework               : @ionic/angular 4.1.2
   @angular-devkit/build-angular : 0.13.7
   @angular-devkit/schematics    : 7.2.4
   @angular/cli                  : 7.3.7
   @ionic/angular-toolkit        : 1.4.1

Cordova:

   cordova (Cordova CLI) : 9.0.0
   Cordova Platforms     : android 8.0.0, ios 5.0.0
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 3.1.2, (and 6 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/johantheitguy/Library/Android/sdk)
   ios-sim           : 7.0.0
   NodeJS            : v10.15.3 (/Users/johantheitguy/.nvm/versions/node/v10.15.3/bin/node)
   npm               : 6.4.1
   OS                : macOS Mojave
   Xcode             : Xcode 10.2 Build version 10E125

### Expected behavior

getProducts should return the list of products.

### Observed behavior

Neither the "then" or "catch" callbacks are ever called.

### Steps to reproduce

Create a blank, new ionic 4 app with similar versions as per above. Copy the demo code for IAP into the app.

**Comment by NassB** [comment]
You are not using the correct plugin. The link https://ionicframework.com/docs/native/in-app-purchase is linked to AlexDisler plugin https://github.com/AlexDisler/cordova-plugin-inapppurchase. But here you need to use https://ionicframework.com/docs/native/in-app-purchase-2 that is linked to j3k0 plugin https://github.com/j3k0/cordova-plugin-purchase.

**Comment by johanvdb** [comment]
Thanks for the feedback. By the time I had posted this I had lost a lot of sleep over the challenge. Neither IAP nor IAP2 wanted to work though. Through a lot of digging I finally found out that there were version clashes in package-lock.json which caused all the drama. Deleting it, node_modules, plugins, and platforms and rebuilding finally saw it working. Hope this at least helps someone in future...
